src/Dialogs/StripeDialog.vala

```diff
@@ -322,7 +322,7 @@ public class AppCenter.Widgets.StripeDialog : Gtk.Dialog {
         try {
             switch (entry) {
                 case 0:
-                    var regex = new Regex ("""[a-z|0-9]+@[a-z|0-9]+\.[a-z]+""");
+                    var regex = new Regex ("""([0-9a-zA-Z]([-\.\+\w]*[0-9a-zA-Z])*@([0-9a-zA-Z][-\w]*[0-9a-zA-Z]\.)+[a-zA-Z]{2,9})""");
```

Comment:
Actually, we should change it for the one suggested here instead of making our own regex :sweat_smile:  https://regular-expressions.mobi/email.html?wlr=1

```suggestion
                    var regex = new Regex ("""[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?""");
```

Reply:
If you're going to use an elaborate version, it should take [all the scenarios](https://en.wikipedia.org/wiki/Email_address#Syntax) in to account. There are a few things wrong with this one, for example:
- forward slashes are not escaped
- it does not allow the local-part to end in capitals (e.g. `my.name.MIT@domain.com`)
- it does not allow capitals in the domain part (e.g. `my.name@Domain.com`)
- it allows an invalid local-part (e.g. `my..name@domain.com`)
- it does not allow the valid version of the one above (e.g. `"my..name"@domain.com`)
- it does not allow comments (e.g. `my.name(comment)@domain.com`)

The only question is, will Stripe accept all those characters as well? Because if that is not the case, there is no use of allowing those exotic email addresses. I'm out all day but tomorrow I'll take a look into it to be sure all scenarios are covered correctly.

Reply:
I just contacted the stipe developers on freenode and as it tuns out, their email validation is quite loose:

> I think we do something quite basic: no space, an @, an extension, and that's mostly it

So instead of including every possible scenario, I think it would be better to go in the opposite direction and follow the same approach as Stripe: keep it simple. So then I would suggest `[^\s]+@[^\s]+\.[^\s]+` if that's ok with you. That's exactly what their developer suggests (no space, an @, an extension).